src/l500/l500-options.cpp

```diff
@@ -571,34 +595,37 @@ namespace librealsense
 
     void l500_options::change_preset( rs2_l500_visual_preset preset )
     {
-        // we need to reset the controls before change gain because after moving to auto gain APD is
-        // read only. This will tell the FW that the control values are defaults and therefore can
-        // be overriden automatically according to gain
-        if( preset == RS2_L500_VISUAL_PRESET_AUTOMATIC )
-        {
-            auto curr_preset = ( rs2_l500_visual_preset )(int)_preset->query();
-            if( curr_preset == RS2_L500_VISUAL_PRESET_AUTOMATIC )
-                return;
-            reset_hw_controls();
-        }
+        // Keep the USB power on while triggering multiple calls on it.
+        ivcam2::group_multiple_fw_calls( get_depth_sensor(), [&]() {
+            // we need to reset the controls before change gain because after moving to auto gain
+            // APD is read only. This will tell the FW that the control values are defaults and
+            // therefore can be overridden automatically according to gain
+            if( preset == RS2_L500_VISUAL_PRESET_AUTOMATIC )
+            {
+                auto curr_preset = ( rs2_l500_visual_preset )(int)_preset->query();
```

Comment:
This doesn't actually go to the FW... the grouping can happen only inside `reset_hw_controls`?

Reply:
I wrapped the whole function because there are several call to the FW, right after _preset->query();

change_gain
change_alt_ir
set_preset_controls_to_defaults
change_laser_power